media-gfx/netgen/netgen-6.2.2204.ebuild

```diff
@@ -0,0 +1,152 @@
+	gui? (
+		dev-lang/tcl:0/8.6
+		dev-lang/tk:0/8.6
+		media-libs/glu
+		virtual/opengl
+		x11-libs/libX11
+		x11-libs/libXmu
+		x11-libs/libxcb:=
+	)
+	jpeg? ( media-libs/libjpeg-turbo:0= )
+	logging? ( dev-libs/spdlog:= )
+	mpi? (
+		sci-libs/metis
+		virtual/mpi
+	)
+	opencascade? ( sci-libs/opencascade:= )
+	python? (
+		${PYTHON_DEPS}
+		$(python_gen_cond_dep '
+			dev-python/pybind11[${PYTHON_USEDEP}]
+			'
+		)
+		mpi? (
+			$(python_gen_cond_dep 'dev-python/mpi4py[${PYTHON_USEDEP}]' )
+		)
+	)
+"
+RDEPEND="${DEPEND}"
+BDEPEND="
+	sys-apps/lsb-release
+	virtual/pkgconfig
+	test? (
+		<dev-cpp/catch-3:0
+		python? ( $(python_gen_cond_dep 'dev-python/pytest[${PYTHON_USEDEP}]') )
+	)
+"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-6.2.2204-use-external-pybind11.patch"
+	"${FILESDIR}/${PN}-6.2.2204-find-Tk-include-directories.patch"
+	"${FILESDIR}/${PN}-6.2.2204-find-libjpeg-turbo-library.patch"
+	"${FILESDIR}/${PN}-6.2.2204-link-against-ffmpeg.patch"
+	"${FILESDIR}/${PN}-6.2.2204-use-system-spdlog.patch"
+	"${FILESDIR}/${PN}-6.2.2204-use-system-catch.patch"
+	"${FILESDIR}/${PN}-6.2.2204-disable-failing-tests.patch"
+	"${FILESDIR}/${PN}-6.2.2204-disable-python-tests.patch"
+)
+
+pkg_setup() {
+	use python && python-single-r1_pkg_setup
+}
+
+src_prepare() {
+	# NOTE: need to manually check and update this string on version bumps!
+	cat <<- EOF > "${S}/version.txt" || die
+		v${PV}-0-gde0d706e
+	EOF
+	cmake_src_prepare
+}
+
+src_configure() {
+	local mycmakeargs=(
+		# currently not working in a sandbox, expects netgen to be installed
+		# see https://github.com/NGSolve/netgen/issues/132
+		-DBUILD_STUB_FILES=OFF
+		-DENABLE_UNIT_TESTS=$(usex test)
+		-DINSTALL_PROFILES=OFF
+		-DNG_INSTALL_DIR_CMAKE="$(get_libdir)/cmake/${PN}"
+		-DNG_INSTALL_DIR_INCLUDE="include/${PN}"
+		-DNG_INSTALL_DIR_LIB="$(get_libdir)"
+		-DUSE_CCACHE=OFF
+		# doesn't build with this version
+		-DUSE_CGNS=OFF
+		-DUSE_GUI=$(usex gui)
+		-DUSE_INTERNAL_TCL=OFF
+		-DUSE_JPEG=$(usex jpeg)
+		-DUSE_MPEG=$(usex ffmpeg)
+		# respect users -march= choice
+		-DUSE_NATIVE_ARCH=OFF
+		-DUSE_MPI=$(usex mpi)
+		-DUSE_OCC=$(usex opencascade)
+		-DUSE_PYTHON=$(usex python)
+		-DUSE_SPDLOG=$(usex logging)
+		-DUSE_SUPERBUILD=OFF
+	)
+	# no need to set this, if we only build the library
+	if use gui; then
+		mycmakeargs+=( -DTK_INCLUDE_PATH="/usr/$(get_libdir)/tk8.6/include" )
+	fi
+	if use python; then
+		mycmakeargs+=(
+			-DPYBIND_INCLUDE_DIR="/usr/lib/${EPYTHON}/site-packages/pybind11/include/"
+			-DNG_INSTALL_PYBIND=OFF
+		)
+	fi
+	if use mpi && use python; then
+		mycmakeargs+=( -DUSE_MPI4PY=ON )
+	else
+		mycmakeargs+=( -DUSE_MPI4PY=OFF )
+	fi
+	cmake_src_configure
+}
+
+src_install() {
+	cmake_src_install
+	use python && python_optimize
+
+	local NETGENDIR="/usr/share/${PN}"
+	echo -e "NETGENDIR=${NETGENDIR}" > ./99netgen || die
+	doenvd 99netgen
+
+	if use gui; then
+		mv "${ED}"/usr/bin/{*.tcl,*.ocf} "${ED}${NETGENDIR}" || die
+
+		doicon "${FILESDIR}"/${PN}.png
```

Comment:
Are we legally allowed to distribute and mirror this icon file? Sometimes images or logos are distributed under different licenses or are subject to trademarks.

Reply:
Don't know. Need to check this.

Reply:
They have OSX and Windows icon files on their repo, but I haven't found one usable for Unix systems.
The file was added back in 2008 according to `git blame`. IMO we **shouldn't** use it without a go from upstream.
I asked upstream at https://github.com/NGSolve/netgen/issues/147.

Could we use the `netgen.ico` for Windows or the `netgen.icns` for OSX? Does freedesktop.org allow for and recognize such files?

Reply:
> Could we use the `netgen.ico` for Windows or the `netgen.icns` for OSX? Does freedesktop.org allow for and recognize such files?

We can't use the `.ico` directly but we can use the `convert` command from `virtual/imagemagick-tools` to "unpack" the `.ico` file into a bunch of differently sized `.png` files and `doicon -s ..` those. See here for an example: https://github.com/gentoo/guru/blob/master/games-action/technic-launcher/technic-launcher-4.625.ebuild

There is also `media-gfx/icoutils` but I have never used it and it is probably better to use imagemagick/graphicsmagick since it is likely already installed on the users system.

Reply:
> We can't use the `.ico` directly but we can use the `convert` command from `virtual/imagemagick-tools` to "unpack" the `.ico` file into a bunch of differently sized `.png` files and `doicon -s ..` those. See here for an example: https://github.com/gentoo/guru/blob/master/games-action/technic-launcher/technic-launcher-4.625.ebuild

I will test whether we can use this.

Reply:
The `.ico` contains 2 usable sizes, 16 and 32. Also droppend the `.desktop` file and generated one using `make_desktop_entry`.